File: tests/fixtures/vulnerable/path_traversal_test.py
# aiscan test fixture — should trigger AI-SEC-011 (Path Traversal)
# DO NOT USE IN PRODUCTION

import os
from pathlib import Path
from flask import request

# Unsafe: user input directly in open()
filename = request.args.get('file')
with open('/var/data/' + filename) as f:
    content = f.read()

# Unsafe: Path() with user-controlled input
user_path = request.query_string
data = Path(user_path).read_text()

# Unsafe: os.path.join with request param
safe_base = '/var/www/static'
full_path = os.path.join(safe_base, request.args.get('path'))

File: tests/fixtures/vulnerable/path_traversal_test.js
// aiscan test fixture — should trigger AI-SEC-011 (Path Traversal, JS/TS)
// DO NOT USE IN PRODUCTION

const fs = require("fs");
const path = require("path");

// Unsafe: fs read with HTTP request body data
app.post("/upload", (req, res) => {
  fs.readFileSync(req.body.path);
});

// Unsafe: path.join with request query parameter
app.get("/file", (req, res) => {
  const full = path.join("/var/data", req.query.name);
  fs.readFileSync(full);
});

// Unsafe: path.resolve with route param
app.get("/asset/:name", (req, res) => {
  const full = path.resolve("/var/assets", req.params.name);
  res.sendFile(full);
});
